src/screens/DiscoveryScreenV2/elements/RecentSearch.js

```diff
@@ -20,100 +20,106 @@ import { fonts } from '../../../utils/fonts';
+
+    const itemIndex = resultArray.indexOf(search);
+    resultArray = [search].concat(resultArray);
+    resultArray.splice(itemIndex + 1, 1);
+    DiscoveryAction.setDiscoveryRecentSearch(resultArray, discoveryDispatch);
+    AsyncStorage.setItem(RECENT_SEARCH_TERMS, JSON.stringify(resultArray));
+    setItems(resultArray);
+  };
+
+  const fetchDiscoveryData = async (search) => {
+    setSearchText(search);
+    setIsFirstTimeOpen(false);
+    Keyboard.dismiss();
+    manipulateSearchTermsOrder(search);
+  };
+
+  const handleClearRecentSearch = async () => {
+    setIsShown(false);
+    // Remove from context here
+    await AsyncStorage.removeItem(RECENT_SEARCH_TERMS);
+
+    DiscoveryAction.setDiscoveryRecentSearch([], discoveryDispatch);
+  };
+
+  if (isShown && items.length > 0)
+    return (
+      <View style={styles.recentContainer}>
         <View style={styles.recentTitleContainer}>
-            <Text style={styles.recentTitle}>Recent</Text>
-            <TouchableOpacity delayPressIn={0} style={styles.iconClear} onPress={handleClearRecentSearch}>
-                <IconClear fill={colors.black} />
-            </TouchableOpacity>
+          <Text style={styles.recentTitle}>Recent</Text>
+          <TouchableOpacity
+            delayPressIn={0}
+            style={styles.iconClear}
+            onPress={handleClearRecentSearch}>
+            <IconClear fill={COLORS.black} />
+          </TouchableOpacity>
         </View>
-        {items.map((item, index) => <RecentSearchItems key={`recentSearch-${index}`} text={item} onItemClicked={() => fetchDiscoveryData(item)} />)}
-    </View>
-
-    return <></>
-
-}
+        {items.map((item, index) => (
+          <RecentSearchItems
+            key={`recentSearch-${index}`}
+            text={item}
+            onItemClicked={() => fetchDiscoveryData(item)}
+          />
+        ))}
+      </View>
+    );
+
+  return <></>;
+};
 
 const styles = StyleSheet.create({
-    iconClear: {
-        alignSelf: 'center',
-        color: colors.black
-    },
-    recentContainer: {
-        paddingTop: 18,
-        paddingBottom: 15,
-        backgroundColor: colors.white
-    },
-    recentTitle: {
-        fontSize: 14,
-        fontFamily: fonts.inter[600],
-        color: colors.bondi_blue,
-        alignSelf: 'center',
-        flex: 1,
-    },
-    recentTitleContainer: {
-        display: 'flex',
-        flexDirection: 'row',
-        paddingLeft: 20,
-        paddingRight: 14,
-        marginBottom: 9,
-    },
-})
+  iconClear: {
+    alignSelf: 'center',
+    color: COLORS.black
+  },
+  recentContainer: {
+    paddingTop: 18,
+    paddingBottom: 15,
+    backgroundColor: COLORS.white
+  },
+  recentTitle: {
+    fontSize: 14,
+    fontFamily: fonts.inter[600],
+    color: COLORS.blue,
+    alignSelf: 'center',
+    flex: 1
+  },
+  recentTitleContainer: {
+    display: 'flex',
+    flexDirection: 'row',
+    paddingLeft: 20,
+    paddingRight: 14,
+    marginBottom: 9
+  }
+});
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In line 102, you are setting color for the `iconClear` style which is used for a `TouchableOpacity`. The `color` property does not have any effect on this component. You should remove this line as it's not doing anything.
```diff
 const styles = StyleSheet.create({
  iconClear: {
    alignSelf: 'center',
-    color: COLORS.black
  },
  recentContainer: {
    paddingTop: 18,
    paddingBottom: 15,
    backgroundColor: COLORS.white
  },
  recentTitle: {
    fontSize: 14,
    fontFamily: fonts.inter[600],
    color: COLORS.blue,
    alignSelf: 'center',
    flex: 1
  },
  recentTitleContainer: {
    display: 'flex',
    flexDirection: 'row',
    paddingLeft: 20,
    paddingRight: 14,
    marginBottom: 9
  }
});
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->